managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -354,6 +354,10 @@ public Map<String, String> getCursorProperties() {
     private CompletableFuture<Void> computeCursorProperties(
             final Function<Map<String, String>, Map<String, String>> updateFunction) {
         CompletableFuture<Void> updateCursorPropertiesResult = new CompletableFuture<>();
+        if (!isDurable()) {
+            updateCursorPropertiesResult.complete(null);
+            return updateCursorPropertiesResult;
+        }
```

Comment:
This doesn't seem to be complete. 
I'd expect that this would happen later, after `newProperties` is set and then applying the changes, something like this:
```
        if (!isDurable()) {
            ManagedCursorImpl.this.cursorProperties = Collections.unmodifiableMap(newProperties);
            updateCursorPropertiesResult.complete(null);
            return updateCursorPropertiesResult;
        }
```

Why didn't the test case catch this issue?